frontend/controller/service-worker.js

```diff
@@ -199,61 +198,51 @@ sbp('sbp/selectors/register', {
       }
 
       const permissionState = await registration.pushManager.permissionState({ userVisibleOnly: true })
+      const granted = permissionState === 'granted' || Notification.permission === 'granted'
+      console.info(`[service-worker/setup-push-subscription] setup: notifications (${notificationEnabled}) perms (${granted})`)
 
-      // Safari sometimes incorrectly reports 'prompt' when using
-      // `registration.pushManager.permissionState`.
-      const existingSubscription = permissionState === 'granted' || Notification.permission === 'granted'
-        ? await registration.pushManager.getSubscription().then((subscription) => {
-          if (
-            !subscription ||
-            (subscription.expirationTime != null && subscription.expirationTime <= Date.now())
-          ) {
-            console.info('[service-worker/setup-push-subscription] attempting to create a new subscription', subscription)
-            return sbp('push/getSubscriptionOptions').then(function (options) {
-              return registration.pushManager.subscribe(options)
-            })
-          } else {
-            const subClone = cloneDeep(subscription)
-            if (subscription?.endpoint) {
-              // hide the full endpoint from the logs for privacy
-              subClone.endpoint = new URL(subClone.endpoint).host
-            }
-            console.info('[service-worker/setup-push-subscription] got existing subscription:', subClone)
+      // Safari sometimes incorrectly reports 'prompt' when using `registration.pushManager.permissionState`
+      let subscription = null
+      let subPrivVer = subscription
+      if (notificationEnabled && granted) {
+        try {
+          subscription = await registration.pushManager.getSubscription()
+          let newSub = false
+          if (!subscription || (subscription.expirationTime != null && subscription.expirationTime <= Date.now())) {
+            subscription = await registration.pushManager.subscribe(await sbp('push/getSubscriptionOptions'))
+            newSub = true
+          }
+          subPrivVer = cloneDeep(subscription)
+          if (subscription?.endpoint) {
+            // hide the full endpoint from the logs for privacy
+            subPrivVer.endpoint = new URL(subPrivVer.endpoint).host
```

Comment:
There are other sensitive things in the subscription info besides the endpoint. Probably the entire `subPrivVer` thing needs to go.